mcmd/commands/make.py

```diff
@@ -56,87 +67,125 @@ def add_arguments(subparsers):
+    url = api.member(group.name)
+    post(url, data={'username': user.username, 'roleName': role.name})
 
 
-def _find_group(role_name: str) -> str:
-    io.debug('Fetching groups')
-    response = get(api.rest2('sys_sec_Role'),
-                   params={
-                       'attrs': 'group(name)',
-                       'q': 'name=={}'.format(role_name)
-                   })
+def _update_group_role_membership(user: User, group: Group, role: Role):
+    io.start('Making user {} a member of role {}'.format(highlight(user.username), highlight(role.name)))
+    url = urljoin(api.member(group.name), user.username)
+    put(url, data=json.dumps({'roleName': role.name}))
+
+
+def _add_role_membership(user: User, role: Role):
+    """
+    Adds a membership manually because the identities API can't add memberships to non-group roles.
+    """
+    io.start('Making user {} a member of role {}'.format(highlight(user.username), highlight(role.name)))
+    membership = {'user': user.id,
+                  'role': role.id,
+                  'from': timestamp()}
+    data = {'entities': [membership]}
+    post(api.rest2('sys_sec_RoleMembership'), data=data)
+
+
+def _get_group_membership(user: User, group: Group) -> Optional[RoleMembership]:
+    group_roles = _get_group_roles(group)
+    group_role_ids = [role.id for role in group_roles]
+
+    memberships = get(api.rest2('sys_sec_RoleMembership'),
+                      params={
+                          'attrs': 'id,user(id,username),role(id,name,label,group(id,name))',
+                          'q': 'user=={};role=in=({})'.format(user.id, ','.join(group_role_ids))
+                      }).json()['items']
 
-    if len(response.json()['items']) == 0:
-        raise McmdError('Role {} not found'.format(role_name))
+    if len(memberships) == 0:
+        return None
+    else:
+        return map_to_role_membership(memberships[0])
 
-    role = response.json()['items'][0]
 
-    if 'group' not in role:
-        raise McmdError('Role {} is not a group role'.format(role_name))
+def _get_group_roles(group: Group) -> List[Role]:
+    roles = get(api.rest2('sys_sec_Role'),
+                params={
+                    'attrs': 'id,name,label,group(id,name)',
+                    'q': 'group=={}'.format(group.id)
+                }).json()['items']
 
-    return role['group']['name']
+    if len(roles) == 0:
+        raise McmdError('No roles found for group {}'.format(group.name))
 
+    return [map_to_role(role) for role in roles]
 
-def _get_user_group_membership(user: str, group_name: str) -> Optional[Membership]:
-    io.debug('Checking if user is already member of this group')
 
-    for membership in _get_memberships_for_user(user):
-        if membership.group_name == group_name:
-            return membership
+def _is_member(user: User, role: Role) -> bool:
+    memberships = get(api.rest2('sys_sec_RoleMembership'),
```

Comment:
I believe role membership has a from-to date? The api is probably more robust way to query this